src/ast/binary-expression.ts

```diff
@@ -0,0 +1,18 @@
+import Context from '../semantics/context';
+import { AstNode } from '../type-definitions/plainscript-types';
+
+export default class BinaryExpression extends AstNode<BinaryExpression> {
+  constructor(public op: any, public left: any, public right: any) { super(); }
```

Comment:
left can be Expression and right can be Expression and op might be String

Reply:
good point ! done !